### DISCUSSION TITLE
How to add padding, margin and other css properties dynamically?

### DISCUSSION THREAD

**Discussion by lalit2005** [mantinedev/mantine#63] [Q&A] [answered]
Manitine is just incredible and the docs are also so slick.
As soon as I saw the components of Mantine, I decided to switch to Mantine from Chakra UI (Chakra is good too 😜 )

In Chakra UI, we can dynamically add all the css properties to components, for example - to add margin and padding of 10px and background color of #eb4034
```jsx
<Box m="10px" p="10px" bgColor="#eb4034">
    <Text>Hello World</Text>
</Box>
```

Can you please tell how can I achieve this in Mantine. I was not able to find any docs related to this. 😞 
**Thanks in advance :)**

**Comment by cstrat** [comment]
The simplest way would be to do this:

````
<Box style={{ margin: '10px', padding: '10px', backgroundColor: '#eb4034' }}>
    <Text>Hello World</Text>
</Box>
`````

You could also use `jss`. I have only tried this a couple of times, looks pretty cool but I haven't needed to deviate from the standard styles much so just use the above.

**Comment by rtivital** [comment, marked as answer]
Hi, you can add styles with `style` prop just like in any other component or element, this prop always have highest priority. You can also use `className` and add your styles with any css solution (jss is not required).

**Reply by lalit2005** [reply]
Yeah that's possible too, but are there any predefined props for adding styles, like how Chakra UI has?
Eg. For margin in x axis, do we have any prop for every component such as `mx` in Mantine?

**Reply by cstrat** [reply]
In another ui-library I use (evergreen) - they have implemented [ui-box](https://github.com/segmentio/ui-box) to do what you're talking about. It actually is pretty handy.

**Reply by lalit2005** [reply]
Or can we use Tailwind CSS with Mantine?? 🤔

**Reply by Paraxdev** [reply]
> Or can we use Tailwind CSS with Mantine?? 🤔

yeah @rtivital how compatible is mantine with tailwind?
for me its mostly working with exceptions like Text component does not change when adding something like "text-9xl2

**Reply by rtivital** [reply]
By default all components support className and style tags, everything should work fine, there might be some css specificity issues due to style tag usage in some components, refer to documentation or source code to see how to control these values
